html/rendering/widgets/baseline-alignment-and-overflow.tentative.html

```diff
@@ -0,0 +1,165 @@
+<!DOCTYPE html>
+<title>HTML: widgets' baseline alignment and interaction with 'overflow'</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<style>
+.test {
+  border-collapse: collapse;
+  font-size: 10px;
+}
+.test td {
+  border: none;
+  padding: 0;
+  margin:0;
+  outline: 1px solid silver;
+}
+
+.test td > input,
+.ref td > .fake-input-text,
+.ref td > .inline-block,
+.ref td > img,
+.ref td > button {
+  font: inherit;
+  height: 60px;
+  width: 60px;
+  padding: 10px;
+  box-sizing: border-box;
+  margin: 10px 0;
+}
+.ref button img {
+  height: 100%;
+  width: 100%;
+  display: block;
+}
+.ref .fake-input-text {
+  display: inline-grid;
+  border: 2px solid;
+  align-items: center;
+}
+.ref .inline-block {
+  display: inline-block;
+}
+.ref td > img.no-margin-bottom {
+  margin-bottom: 0;
+}
+[style*="appearance: none;"] {
+  -webkit-appearance: none; /* TODO(zcorpan) remove this when unprefixed appearance is supported */
+}
+</style>
+<div id="log"></div>
+<h2>refs</h2>
+<!--
+The first span's offsetTop is what we want to compare with the corresponding test's span's offsetTop.
+The sibling element is there to control where the baseline for the line box will be.
```

Comment:
It's still not really clear to me why, for example, the `offsetTop` of a span that sits next to an `<input type=range>` should be the same as that of a span that sits next to an `<img>` with a valid `src=""` attribute (`refImgSrcOffsetTop`).

Or why the `offsetTop` of a span that sits next to an `<input type=checkbox>` should be the same as that of a span that sits next to an  `<img>` with a valid `src=""` attribute and no margin-bottom (which is for some reason called `refCheckboxOffsetTop` instead of `refImgSrcNoMarginBottomOffsetTop`).

Or why the `offsetTop` of a span that sits next to an `<input type=tel>` should be the same as that of a span that sits next to another span which uses `display: inline-grid; border: 2px solid; align-items: center;`.

In other words, where do these elements on the right, like img with src, or img with src and no margin bottom, come from? Does this correspond to some CSS concept?

Maybe this would become clearer if there were a spec. Probably the plan is not to write the spec as something literally like "the `offsetTop` for a span that sits next to an `<input type=range>` must match the `offsetTop` of a span that sits next to an `<img>` with a valid `src=""` attribute"? What would the spec be? I gather from the test title that we're testing "baseline alignment", and I guess `offsetTop` is a way of measuring that?

Reply:
The spec would say where the baseline is for each widget. See https://github.com/whatwg/html/issues/5065

I've added a comment trying to explain what's going on with this test.

Reply:
> (which is for some reason called `refCheckboxOffsetTop` instead of `refImgSrcNoMarginBottomOffsetTop`)

The variable names are intended to say which widget they emulate, not how the fake widget is implemented.

Reply:
> The spec would say where the baseline is for each widget. See whatwg/html#5065

This issue didn't clarify anything for me. It says that the spec should specify a baseline. (Thankfully the new comment you added clarifies that this is related to offsetTop.) It doesn't say what that baseline will be, or why type=tel should correspond to span with display: inline-grid, border 2px solid, align-items center.

> The variable names are intended to say which widget they emulate, not how the fake widget is implemented.

Is the idea that a span with display: inline-grid, border 2px solid, align-items center is something like a text widget? How is the text widget/span-with-those-properties category, or the checkbox widget/img-with-valid-src-and-no-margin-bottom category, related to the four categories you list in lines 24-27?

Will the spec eventually talk about those four categories, or will it talk about spans-with-those-properties, or...? Why are there 6 refs, but 4 categories?

Reply:
> Is the idea that a span with display: inline-grid, border 2px solid, align-items center is something like a text widget?

Yes.

> How is the text widget/span-with-those-properties category, or the checkbox widget/img-with-valid-src-and-no-margin-bottom category, related to the four categories you list in lines 24-27?

I've clarified this now.

> Will the spec eventually talk about those four categories, or will it talk about spans-with-those-properties, or...? Why are there 6 refs, but 4 categories?

The new comment could be used as a starting point for writing the spec, I think!

Thanks for pushing, here. A test that is confusing isn't helping anyone. 🙂